src/astro/utils/cloud_storage_creds.py

```diff
@@ -15,24 +18,36 @@ def parse_s3_env_var():
     return [parse.unquote(r) for r in raw_data]
 
 
-def s3fs_creds():
-    # To-do: reuse this method from sql decorator
+def s3fs_creds(conn_id=None):
     """Structure s3fs credentials from Airflow connection.
     s3fs enables pandas to write to s3
     """
-    # To-do: clean-up how S3 creds are passed to s3fs
+    if conn_id:
+        extra = BaseHook.get_connection(conn_id).extra_dejson
+        key = extra["aws_access_key_id"]
```

Comment:
@tatiana  - Are we assuming below connection type for aws connection?
https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/connections/aws.html

Reply:
@utkarsharma2 Since the [S3Hook inherits](https://airflow.apache.org/docs/apache-airflow/1.10.6/_modules/airflow/hooks/S3_hook.html#S3Hook) from the base AWS Hook,  I tried to make it consistent with examples I found and [one of the currently supported ways of authenticating](https://airflow.apache.org/docs/apache-airflow-providers-amazon/1.1.0/_modules/airflow/providers/amazon/aws/hooks/base_aws.html) - so we could tie it to `smart_open`. Given the scope of this ticket, do you think we should change this implementation?

Reply:
@tatiana, can we have something like this - https://github.com/apache/airflow/blob/99cf75d5fd07aad7db0a4290b1c7663485e999af/airflow/providers/amazon/aws/hooks/base_aws.py#L200 

or better if we can use this somehow?

Reply:
If we just finalize approach here I can pick in my end of this task.

Reply:
@utkarsharma2 the `BaseSessionFactory` isn't available in Airflow `2.2.4`, but I found another way - please, check it out!

Reply:
LGTM.